lib/util.js

```diff
@@ -1022,24 +1027,27 @@ class Load {
    * @return {Object} - The module level configuration object.
    */
   setModuleDefaults(moduleName, defaultProperties) {
-    const moduleConfig = Util.cloneDeep(defaultProperties);
-    let moduleDefaults;
+    if (this.defaults === undefined) {
```

Comment:
I’ll add some explanations for posterity since this code seems to be a bit of a chore to read. 

This is a zero cost abstraction. To fix the bugs required more memory footprint for config. So don’t allocate the extra storage if the feature is not being used.